src/intersections/boxes.jl

```diff
@@ -34,3 +34,59 @@ function intersecttype(f::Function, b1::Box{Dim,T}, b2::Box{Dim,T}) where {Dim,T
     return FaceTouchingBoxes(Box(u, v)) |> f
   end
 end
+
+"""
+    intersecttype(function, ray, box)
+
+Calculate the intersection type of a `ray`` and a `box`` and apply function `f` to it.
+
+## References
+
+* Williams A, Barrus S, Morley R K, et al., 2005. [An efficient and robust ray-box
+  intersection algorithm]
+  (https://dl.acm.org/doi/abs/10.1145/1198555.1198748)
+"""
+function intersecttype(f::Function, r::Ray{3,T}, b::Box{3,T}) where {T}
+  sign = (x -> x < 0 ? 2 : 1).(direction(r))
```

Comment:
This variable is only used later. Please define variables near where they are used. Also, this syntax you are using is not aligned with our code style, please consider removing the lambda. Let's start with the version of the algorithm that is more natural, and then we can consider performance optimizations. You are mixing these two things.